### PR TITLE
controller: Increment log

### PR DESCRIPTION
**Pull request by Igoraamc**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This pull request implements unit tests for all interactor features.

- Adds new logs in controller

Does this close any currently open issues?
------------------------------------------
Any relevant logs, error output, etc?
-------------------------------------
Any other comments?
-------------------
Where has this been tested?
---------------------------

**Operating System/Platform:** MACOS Catalina

**GO Version:** go 1.13.7 darwin/amd64

### CODE REVIEW

**Review by esdrasjnr — APPROVED** [review]
Excellent! The `error` logs looks awesome to me. However, I'm not sure about the use of `debug` and `info` logs, and the difference between them both
Wait to see what @netoax and @lcbm comment about! :smile:

File: pkg/controllers/data.go [hunk 1]
```diff
@@ -49,64 +49,79 @@ func (d *DataController) writeResponse(w http.ResponseWriter, statusCode int, ms
...
 	token := r.Header.Get("auth_token")
 
 	err := d.DataInteractor.Delete(token, deviceID)
 	if err != nil {
+		d.logger.Errorf("failed to delete data: %s", err)
 		d.writeResponse(w, http.StatusUnprocessableEntity, "Invalid information")
 		return
 	}
 
+	d.logger.Info("successfully deleted data")
```
Comment (lcbm) [inline pkg/controllers/data.go]: Could we have richer logs? This is a good example because we could add the `deviceId` which was deleted, for example.

**Review by lcbm — COMMENTED** [review]
LGTM! About the log levels, form what I understand it's fine. I would try to add more context to the log, but that's just a suggestion. For this reason, I'm neither approving nor requesting changes. Let me know your thoughts! :smile:

File: pkg/controllers/data.go (hunk 1 above)
Comment (netoax) [reply pkg/controllers/data.go]: +1 on this.

File: pkg/controllers/data.go [hunk 2]
```diff
@@ -49,64 +49,79 @@ func (d *DataController) writeResponse(w http.ResponseWriter, statusCode int, ms
 
 // List handles incoming data list requests
 func (d *DataController) List(w http.ResponseWriter, r *http.Request) {
+	d.logger.Debug("get data request received")
```
Comment (netoax) [inline pkg/controllers/data.go:52]: Do you know if request parameters are already being logged (by the `mux`, for example)? I think for debug level it'd be useful.

File: pkg/controllers/data.go [hunk 3]
```diff
@@ -49,64 +49,79 @@ func (d *DataController) writeResponse(w http.ResponseWriter, statusCode int, ms
...
 
 	token := r.Header.Get("auth_token")
 	data, err := d.DataInteractor.List(token, query)
 	if err != nil {
+		d.logger.Errorf("failed to get data: %s", err)
 		d.writeResponse(w, http.StatusInternalServerError, err.Error())
 		return
 	}
 
+	d.logger.Info("successfully got data")
```
Comment (netoax) [inline pkg/controllers/data.go]:
```suggestion
	d.logger.Info("data successfully obtained")
```

**Review by netoax — CHANGES_REQUESTED** [review]
LGTM, thanks!
I did just some comments. Please, consider the suggestion @lcbm made also.

File: pkg/controllers/data.go (hunk 3 above)
Comment (Igoraamc) [reply pkg/controllers/data.go]: Done!

File: pkg/controllers/data.go (hunk 2 above)
Comment (Igoraamc) [reply pkg/controllers/data.go:52]: Actually, mux logs the requested endpoint, including the query parameters. But, either way, I added a log for the query (on debug mode).

File: pkg/controllers/data.go (hunk 1 above)
Comment (Igoraamc) [reply pkg/controllers/data.go]: I totally agree with you, guys. As @lcbm suggested, I added the deviceID passed by the user in the log. I'll find other logs that can be richer.

**Comment by netoax** [comment]
> Excellent! The `error` logs looks awesome to me. However, I'm not sure about the use of `debug` and `info` logs, and the difference between them both
> Wait to see what @netoax and @lcbm comment about! 😄

@esdrasjnr We usually use `debug` for more detailed contextual information about the operations is being executed. Besides, `info` just shows that operations were received and executed or not.

**Review by netoax — APPROVED** [review]
LGTM!

**Review by lcbm — APPROVED** [review]
A-w-e-s-o-m-e! :100:

**Pull request merged**